spec/cipherstash/index/bloom_filter_spec.rb

```diff
@@ -3,27 +3,133 @@
 require "cipherstash/index/bloom_filter"
 
 describe CipherStash::Index::BloomFilter do
-  # Generated by SecureRandom.hex(16)
+  # Generated by SecureRandom.hex(32)
   # The same key is used for each test run so that these tests are deterministic.
-  let(:key) { "8b42157da439e06aa6f6d1ad8c1db72e" }
+  let(:key) { "b6d6dba3be33ffaabb83af611ec043b9270dacdc7b3015ce2c36ba17cf2d3b2c" }
 
   describe ".new" do
     it "returns a bloom filter with empty bits" do
       filter = described_class.new(key)
       expect(filter.bits).to eq(Set.new())
     end
+
+    it "provides a default for m" do
+      filter = described_class.new(key)
+      expect(filter.m).to eq(256)
+    end
+
+    described_class::VALID_M_VALUES.each do |m|
+      it "allows #{m} as a value for m" do
+        filter = described_class.new(key, {"filterSize" => m})
+        expect(filter.m).to eq(m)
+      end
+    end
+
+    [0, 2, 16, 31, 513, 131072].each do |m|
+      it "raises given invalid m #{m}" do
+        expect {
+          described_class.new(key, {"filterSize" => m})
+        }.to raise_error(::CipherStash::Client::Error::InvalidSchemaError, "filterSize must be a power of 2 between 32 and 65536")
+      end
+    end
+
+    it "provides a default for k" do
+      filter = described_class.new(key)
+      expect(filter.k).to eq(3)
+    end
+
+    (3..16).each do |k|
+      it "allows #{k} as a value for k" do
+        filter = described_class.new(key, {"filterTermBits" => k})
+        expect(filter.k).to eq(k)
+      end
+    end
+
+    it "raises when k is < 3" do
+      expect {
+        described_class.new(key, {"filterTermBits" => 2})
+      }.to raise_error(::CipherStash::Client::Error::InvalidSchemaError, "filterTermBits must be between 3 and 16")
+    end
+
+    it "raises when k is > 16" do
```

Comment:
```
it "raises when k is not a float"
it "raises when k is a string"
it "raises when k is a small, furry creature from Alpha Centauri"
```

Reply:
Nice, this helped catch a bug where k could actually be set to a float.